src/common/query-parser/QueryParser.ts

```diff
@@ -0,0 +1,90 @@
+import {astVisitor, ExprRef, FromStatement, FromTable, IAstVisitor, parse, SelectedColumn, SelectFromStatement, WithRecursiveStatement, WithStatement} from "pgsql-ast-parser";
+import { QueryTreeTracker } from "./QueryTreeTracker";
+import { QueryTree } from "./tree/QueryTree";
+
+export class QueryParser {
```

Comment:
1. I wanted to not restrict it to be only used in UI. Hence it is in common.
2. Just a matter to preference. It could be either a class or a function. I prefer having a class for everything.